web/src/engine/osk/src/keyboard-layout/gesturePreviewHost.ts

```diff
@@ -0,0 +1,137 @@
+import { ActiveKey } from "@keymanapp/keyboard-processor";
+import { KeyElement } from "../keyElement.js";
+import { FlickNameCoordMap, OrderedFlickDirections } from "../input/gestures/browser/flick.js";
+
+const FLICK_DIRS = ['n', 'ne', 'e', 'se', 's', 'sw', 'w', 'nw'] as const;
+export class GesturePreviewHost {
+  private readonly div: HTMLDivElement;
+  private readonly label: HTMLSpanElement;
+  private readonly previewImgContainer: HTMLDivElement;
+
+  private flickPreviews = new Map<string, HTMLDivElement>;
+  private hintLabel: HTMLDivElement = null;
+  private flickEdgeLength: number;
+
+  private onCancel: () => void;
+
+  get element(): HTMLDivElement {
+    return this.div;
+  }
+
+  constructor(key: KeyElement, isPhone: boolean, edgeLength: number) {
+    const keySpec = key.key.spec;
+    this.flickEdgeLength = edgeLength;
+
+    const base = this.div = document.createElement('div');
+    base.className='kmw-gesture-preview';
+    base.id = 'kmw-gesture-preview';
+
+    base.style.pointerEvents='none';
+
+    // We want this to be distinct from the base element so that we can scroll it;
+    // this matters greatly for doing flick things.
+    const previewImgContainer = this.previewImgContainer = document.createElement('div');
+    this.previewImgContainer.id = 'kmw-preview-img-container';
+
+    const label = this.label = document.createElement('span');
+    label.className='kmw-gesture-base-label kmw-key-text';
+    label.id = 'kmw-gesture-base-label';
+    previewImgContainer.appendChild(label);
+
+    // Re-use the text value from the base key's label.
+    label.textContent = key.key.label.textContent;
+
+    this.div.appendChild(this.previewImgContainer);
+
+    if(keySpec.flick) {
+      const flickSpec = keySpec.flick || {};
+
+      Object.keys(flickSpec).forEach((dir: typeof OrderedFlickDirections[number]) => {
+        const flickPreview = document.createElement('div');
+        flickPreview.className = 'kmw-flick-preview kmw-key-text';
+        flickPreview.textContent = flickSpec[dir].text;
+
+        const ps /* preview style */ = flickPreview.style;
+
+        const OVERFLOW_OFFSET = 1.41; //141;
```

Comment:
What is this value for? The `//141` explanatory comment is not helping me much :grin:

Reply:
Right, guess I forgot to do cleanup there.  Is sqrt(2) * 100; the CSS is specified on a percentage offset from a border.  Previews for the 'n' flick are offset from the top border by 141%, while 'ne' flicks are offset by ~100% from both the top and left borders.

But yeah, it'd help to explain that as an actual comment.